src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -57,32 +69,37 @@ const TopicFragment = ({
   const handleFollow = async (from, willFollow, item, index) => {
     if (from === FROM_FOLLOWED_TOPIC_INITIAL) {
       const newFollowedTopics = [...topics];
-      newFollowedTopics[index].user_id_follower = willFollow ? myId : null;
+      newFollowedTopics[index].following = !!willFollow;
+      newFollowedTopics[index].user_id_follower = myId;
       DiscoveryAction.setDiscoveryInitialTopics(newFollowedTopics, discoveryDispatch);
     }
 
     if (from === FROM_UNFOLLOWED_TOPIC_INITIAL) {
       const newFollowedTopics = [...topics];
-      newFollowedTopics[index].user_id_follower = willFollow ? myId : null;
+      newFollowedTopics[index].following = !!willFollow;
+      newFollowedTopics[index].user_id_follower = myId;
 
       DiscoveryAction.setDiscoveryInitialTopics(newFollowedTopics, discoveryDispatch);
     }
 
     if (from === FROM_FOLLOWED_TOPIC) {
       const newFollowedTopics = [...followedTopic];
-      newFollowedTopics[index].user_id_follower = willFollow ? myId : null;
+      newFollowedTopics[index].following = !!willFollow;
+      newFollowedTopics[index].user_id_follower = myId;
 
       setFollowedTopic(newFollowedTopics);
     }
 
     if (from === FROM_UNFOLLOWED_TOPIC) {
       const newUnFollowedTopic = [...unfollowedTopic];
-      newUnFollowedTopic[index].user_id_follower = willFollow ? myId : null;
+      newUnFollowedTopic[index].following = !!willFollow;
+      newUnFollowedTopic[index].user_id_follower = myId;
 
       setUnfollowedTopic(newUnFollowedTopic);
     }
 
     followTopic(item?.name);
+    if (searchText.length > 0) fetchData();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleFollow` function has been updated to handle the `following` state of each topic. This change seems to be more intuitive and easier to understand than the previous implementation.
```diff
- newFollowedTopics[index].user_id_follower = willFollow ? myId : null;
+ newFollowedTopics[index].following = !!willFollow;
+ newFollowedTopics[index].user_id_follower = myId;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->